src/ubx.cpp

```diff
@@ -2016,9 +2164,10 @@ GPSDriverUBX::payloadRxDone()
 			float rel_length_acc = _buf.payload_rx_nav_relposned.accLength * 1e-2f;
 			bool heading_valid = _buf.payload_rx_nav_relposned.flags & (1 << 8);
 			bool rel_pos_valid = _buf.payload_rx_nav_relposned.flags & (1 << 2);
+			bool carrier_solution_fixed = _buf.payload_rx_nav_relposned.flags & (1 << 4);
 			(void)rel_length_acc;
 
-			if (heading_valid && rel_pos_valid && rel_length < 1000.f) { // validity & sanity checks
+			if (heading_valid && rel_pos_valid && rel_length < 1000.f && carrier_solution_fixed) { // validity & sanity checks
```

Comment:
@benflyvoly I believe these are the "strange data checks" you mentioned that we might consider removing.

Reply:
Yes. This is something I didn't agree with: Only pass awesome solutions without status flags could lead to all NAN heading without knowing why. Either extend relpos status or remove and rely on heading_accuracy as status are preferred.

Reply:
That sounds good to me; my tests have just been giving me nothing but NaN with no status data.  I'll look into getting this data across the CAN bus to the autopilot somehow.

Reply:
Is this something we should change now, or do we also need to add consistency checks in front of the EKF before removing them here?